src/chunk_store.rs

```diff
@@ -187,22 +188,22 @@ impl ChunkStore {
                 };
                 Ok(dir_entries.filter_map(dir_entry_to_routing_name).collect())
             })
-            .unwrap_or(vec![])
+            .unwrap_or_else(|_| Vec::new())
```

Comment:
Just nitpicking, but don't think this is a good idea. `Vec::new()` is zero allocation, so essentially free. `unwrap_or_else` involves calling the lambda, which might have overhead. Although it's probably going to be optimized away anyway, not sure.